### PR TITLE
Dataframe view update and blueprint API (part 2): UI for the query

### PR DESCRIPTION
**Pull request by abey79**
### What

This PR adds a wrapper over the query view property introduced in #7516 and implement the selection panel UI for it. This UI is functional (e.g. filter event entity and component auto-suggestion, etc.), but it is still ignored by the view itself and is displayed alongside the original (still functional) UI.

image

<hr>

Part of a series to address #6896 and #7498.

All PRs:
- #7515
- #7516
- #7527 
- #7545
- #7551
- #7572
- #7573

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7527?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7527?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7527)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_space_view_dataframe/src/view_query_v2/blueprint_io.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            .query_property
+            .component_or_empty::<components::SelectedColumns>()?;
+
+        // no selected columns means all columns are visible
+        let Some(datatypes::SelectedColumns {
+            time_columns,
+            component_columns,
+        }) = selected_columns.as_deref()
+        else {
+            return Ok(None);
```
Comment (teh-cmc) [inline crates/viewer/re_space_view_dataframe/src/view_query_v2/blueprint_io.rs:157]: I figure this means `None` == all columns? Would be nice to document that in the doc-comment

File: crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+    pub(super) fn filter_range_ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        timeline: &Timeline,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        let time_drag_value = if let Some(times) = ctx.recording().time_histogram(timeline) {
+            TimeDragValue::from_time_histogram(times)
+        } else {
+            // This should never happen because `timeline` is guaranteed to be valid by `Self::timeline()`
```
Comment (teh-cmc) [inline crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs:47]:
```suggestion
            debug_assert!(false, "This should never happen because `timeline` is guaranteed to be valid by `Self::timeline()`");

File: crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+
+        Ok(())
+    }
+
+    pub(super) fn column_visibility_ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        timeline: &Timeline,
+        schema: &[ColumnDescriptor],
```
Comment (teh-cmc) [inline crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs:260]: I really don't like the name `schema` (here and everywhere else) in the context of the new APIs.

We need to be explicit, always, or it's extremely hard to follow what's going on: is this the `view_columns` (which control the view contents) or the `selected_columns` (which control the selection)?

**Review by teh-cmc — APPROVED** [review]

File: crates/viewer/re_space_view_dataframe/src/view_query_v2/blueprint_io.rs
[the remark this answers is not in the document]
Comment (abey79) [reply crates/viewer/re_space_view_dataframe/src/view_query_v2/blueprint_io.rs:1]: Yeah, I meant it as read-writes to the blueprint store. I'll rename it blueprint.rs

File: crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs (hunk 3 above)
Comment (abey79) [reply crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs:260]: That make sense to me.

`schema` -> always arrow schema
`view_columns` -> all columns in the view (depends on the `view_contents`)
`selected_columns` -> the columns that remains after applying the user selection (`select` arg/func in the API)

This feels like something @jleibs should be on-board with though.

I'm applying some of that here, and will hunt every remaining `schema` in part 6.
